### ISSUE TITLE
Accuracy of the model in <model.fit> and <model.evaluate> does not match the accuracy of the sklearn confusion matrix

### ISSUE THREAD

**Issue by lfluchi** [lfluchi/ring_detection#1]
I am working on this convolutional model for image classification.

The classification by the confusion matrix is always in a single class, but during training and when evaluating with `model.evaluate` the accuracy value is high. I thought it was a problem with the generator so I changed the model so I wouldn't have to use it anymore.

My data is very unbalanced so I oversampled the minority class of the training data. I know that accuracy is not the best metric in this case, but the confusion matrix should show how good the model is, at least in the training data.

The model may be badly dimensioned but how will I improve if the accuracy during training and validation is already very high?
Am I missing something?

Loss and accuracy with `model.evaluate` on trainning data:
Loss, Accuracy: [7.230345488995482e-11, 1.0]

Classification report on trainning data:
precision    recall  f1-score   support

        boaz       0.50      1.00      0.67      6068
       ruinz       0.00      0.00      0.00      6066

    accuracy                           0.50     12134
   macro avg       0.25      0.50      0.33     12134
weighted avg       0.25      0.50      0.33     12134

Complete code is here:
https://github.com/lfluchi/ring_detection/blob/master/CNN_anel_over_s.ipynb

**Comment by lfluchi** [comment]
The scores with `model.evaluate` were correct. I was using `model.predict` in a incorrect manner. When the classification problem is binary and you are using a sigmoid function as a the activation of the last layer you have to separate values above or below 0.5, for example using `predicted_classes = np.round(predictions, decimals=0)` instead of `predicted_classes = numpy.argmax(predictions, axis=1)` (used for multiclass classification with softmax function).
